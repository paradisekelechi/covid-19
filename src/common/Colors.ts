export default {
    darkBlue: '#181634',
    skyBlue: '#DFE4E6',
    white: '#FFFFFF',
    black: '#000000',
    maroon: '#341616',
    darkGreen: '#183416',
    armyGreen: '#333416',
    tourquoisBlue: '#2AA1D1',
    fairBlue: '#8FB8C9',
    paleRed: '#7C5151',
    errorRed: '#9D3535',
    green: '#488010',
    yellowishGreen: '#838D17'
}