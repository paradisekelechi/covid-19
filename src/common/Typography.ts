import styled from 'styled-components';
import Colors from '../common/Colors';

export type ModeType = 'dark' | 'light';

type TitleType = {
  mainHeader: boolean;
  mode: ModeType;
}

export const Title = styled.h3<TitleType>`
  letter-spacing: 0px;
  font-family: ${props => props.mainHeader ? 'Berkshire Swash' : 'Kodchasan'};
  font-size: 38px;
  line-height: 47px;
  color: ${props => props.mode === 'dark' ? Colors.skyBlue : Colors.darkBlue};
  margin: 0px;
`;

type ContentTextType = {
  basic?: boolean;
  bold?: boolean;
  mode: ModeType;
}

export const ContentText = styled.p<ContentTextType>`
  font-family:  'Kodchasan';
  font-size: ${props => props.basic ? '1.1em' : '1.4em'};
  font-weight: ${props => props.bold ? '600' : 'normal'};
  color: ${props => props.mode === 'dark' ? Colors.skyBlue : Colors.darkBlue};
`;

const StatText = styled.h2`
  letter-spacing: 0px;
  font-family: 'Berkshire Swash';
  color: ${Colors.skyBlue};
  margin: 0px;
  width: 100%;
  text-align: center;
`;

export const StatTitleText = styled(StatText)`
  font-size: 3.2em;
`;

export const StatContentText = styled(StatText)`
  font-size: 1.25em;
`;

const ContentUnitText = styled.p<{ color?: string, leftAligned?: boolean }>`
  letter-spacing: 0px;
  font-family: 'Kodchasan';
  color: ${props => props.color};
  margin: 0px;
  width: 100%;
  text-align: ${props => props.leftAligned ? 'left' : 'center'};
`;

export const ContentUnitTitleText = styled(ContentUnitText)`
  font-size: 1.2em;
  font-weight: 600;
`;

export const ContentUnitContentText = styled(ContentUnitText)`
  font-size: 1em;
`;
