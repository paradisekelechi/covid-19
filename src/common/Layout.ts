import styled from 'styled-components';
import { ModeType } from './Typography';
import Colors from './Colors';

type ContainerType = {
    mode?: ModeType;
}

export const Container = styled.div<ContainerType>`
    width: 100%;
    min-height: 100vh;
    background-color: ${props => props.mode && props.mode === 'dark' ? Colors.darkBlue : Colors.skyBlue};
    padding: 50px 100px;
    box-sizing: border-box;
    margin: 0px;
    transition: all 800ms;
`;

export type SpacingType = 'none' | 'small' | 'medium' | 'large';
export type AlignmentType = 'left' | 'right' | 'space-between' | 'space-evenly';

type WrapperType = {
    spacing: SpacingType;
    alignment?: AlignmentType;
    mode?: ModeType;
}

export const BlockWrapper = styled.div<WrapperType>`
    width: 100%;
    box-sizing: border-box;
    margin-top: ${props => {
        switch (props.spacing) {
            case 'none':
                return '0px';
            case 'small':
                return '1em';
            case 'medium':
                return '2em';
            case 'large':
                return '4.5em';

            default:
                return '0px';
        }
    }};
`;


export const Wrapper = styled(BlockWrapper)`
    display: flex;
    justify-content: ${props => {
        switch (props.alignment) {
            case 'left':
                return 'flex-start';
            case 'right':
                return 'flex-end';
            case 'space-between':
                return 'space-between';
            case 'space-evenly':
                return 'space-evenly';
            default:
                return 'flex-start';
        }
    }};
`;

export const Card = styled.div<{ color?: string, mode: ModeType }>`
    width: 22%;
    padding: 35px 0px;
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    box-shadow: 0px 3px 20px ${props => props.mode && props.mode === 'dark' ? Colors.skyBlue : Colors.darkBlue};
    background-color: ${props => props.color};
    border-top: 2px solid ${props => props.mode && props.mode === 'dark' ? Colors.skyBlue : Colors.darkBlue};
`;

export const ContentRow = styled(Wrapper)`
    border: 1px solid ${props => props.mode && props.mode === 'dark' ? Colors.skyBlue : Colors.darkBlue};
    padding: 20px;
`;

export const ContentRowUnit = styled.div`
    width: 12%;
`;



