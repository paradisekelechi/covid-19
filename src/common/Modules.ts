import styled, { keyframes } from 'styled-components';
import { ModeType } from './Typography';
import Colors from './Colors';

type ToggleType = {
    checked: boolean;
}

export const Switch = styled.label`
    position: relative;
    display: inline-block;
    width: 66px;
    height: 40px;
`;

export const SwitchInput = styled.input`
    opacity: 0;
    width: 0;
    height: 0;
`;

export const SwitchSlider = styled.span<ToggleType>`
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: ${props => props.checked ? Colors.skyBlue : Colors.darkBlue};
    border: 2px solid ${props => props.checked ? Colors.darkBlue : Colors.skyBlue};
    -webkit-transition: .4s;
    transition: .4s;

    &:before {
        position: absolute;
        content: "";
        height: 26px;
        width: 26px;
        left: 4px;
        bottom: 4px;
        background-color: ${props => props.checked ? Colors.darkBlue : Colors.skyBlue};
        -webkit-transform: translateX(${props => props.checked ? '26px' : '0'});
        -ms-transform: translateX(${props => props.checked ? '26px' : '0'});
        transform: translateX(${props => props.checked ? '26px' : '0'});
        -webkit-transition: .4s;
        transition: .4s;
    }

    input:focus + &{
        box-shadow: 0 0 1px ${Colors.darkBlue};
    }
`;

export const Flag = styled.img`
`;

const spinLoader = () => keyframes`
    0%{
        transform: rotate(0deg);
    }
    100%{
        transform: rotate(360deg);
    }
`;

export const Loader = styled.div<{ mode: ModeType; }>`
    width: 80px;
    height: 80px;
    border-radius: 50%;
    border: 8px solid ${props => props.mode && props.mode === 'dark' ? Colors.skyBlue : Colors.darkBlue};
    border-top: 8px solid ${Colors.fairBlue};
    animation: ${spinLoader} 1600ms linear infinite;
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
`;