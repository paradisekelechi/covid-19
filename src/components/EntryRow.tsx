import * as React from 'react';
import { ModeType, ContentUnitContentText, ContentUnitTitleText } from '../common/Typography';
import { ContentRow, ContentRowUnit } from '../common/Layout';
import { Flag } from '../common/Modules';
import Colors from '../common/Colors';

type EntryDataType = {
    country: string;
    countryInfo: {
        _id: number;
        iso2: string;
        iso3: string;
        lat: number;
        long: number;
        flag: string;
    },
    cases: number;
    todayCases: number;
    deaths: number;
    todayDeaths: number;
    recovered: number;
    active: number;
    critical: number;
    casesPerOneMillion: number;
    deathsPerOneMillion: number;
    updated: number;
}

export interface EntryRowProps {
    mode: ModeType;
    data: EntryDataType;
}

type getFormattedNumberType = (number: number) => string;

export const getFormattedNumber: getFormattedNumberType = (number) => {
    return new Intl.NumberFormat('en').format(number);
}

const EntryRow: React.SFC<EntryRowProps> = ({ mode, data }) => {
    const { country, countryInfo: { flag }, cases, todayCases, deaths, todayDeaths, active, recovered } = data;
    return (
        <ContentRow spacing='none' alignment='space-evenly' mode={mode as ModeType}>

            <ContentRowUnit >
                <Flag src={flag} width="100" height="44" />
                <ContentUnitTitleText color={Colors.tourquoisBlue} leftAligned={true}>{country}</ContentUnitTitleText>
            </ContentRowUnit>

            <ContentRowUnit >
                <ContentUnitTitleText color={Colors.tourquoisBlue}>{getFormattedNumber(Number(cases))}</ContentUnitTitleText>
                <ContentUnitContentText color={Colors.tourquoisBlue}>Total Cases</ContentUnitContentText>
            </ContentRowUnit>

            <ContentRowUnit >
                <ContentUnitTitleText color={Colors.fairBlue}>{getFormattedNumber(todayCases)}</ContentUnitTitleText>
                <ContentUnitContentText color={Colors.fairBlue}>Cases Today</ContentUnitContentText>
            </ContentRowUnit>

            <ContentRowUnit >
                <ContentUnitTitleText color={Colors.paleRed}>{getFormattedNumber(deaths)}</ContentUnitTitleText>
                <ContentUnitContentText color={Colors.paleRed}>Total Deaths</ContentUnitContentText>
            </ContentRowUnit>

            <ContentRowUnit >
                <ContentUnitTitleText color={Colors.errorRed}>{getFormattedNumber(todayDeaths)}</ContentUnitTitleText>
                <ContentUnitContentText color={Colors.errorRed}>Deaths Today</ContentUnitContentText>
            </ContentRowUnit>

            <ContentRowUnit >
                <ContentUnitTitleText color={Colors.green}>{getFormattedNumber(recovered)}</ContentUnitTitleText>
                <ContentUnitContentText color={Colors.green}>Recovered</ContentUnitContentText>
            </ContentRowUnit>

            <ContentRowUnit>
                <ContentUnitTitleText color={Colors.yellowishGreen}>{getFormattedNumber(active)}</ContentUnitTitleText>
                <ContentUnitContentText color={Colors.yellowishGreen}>Active Cases</ContentUnitContentText>
            </ContentRowUnit>

        </ContentRow>
    );
}

export default EntryRow;