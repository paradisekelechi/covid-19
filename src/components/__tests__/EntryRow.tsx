import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Colors from '../../common/Colors';
import {ModeType} from '../../common/Typography';
import EntryRow from '../EntryRow';
import {getFormattedNumber} from '../EntryRow';

describe('EntryRow', () => {
    const data = {
        "country": "Turkey",
        "countryInfo": {
            "_id": 792,
            "iso2": "TR",
            "iso3": "TUR",
            "lat": 39,
            "long": 35,
            "flag": "https://raw.githubusercontent.com/NovelCOVID/API/master/assets/flags/tr.png"
        },
        "cases": 23934,
        "todayCases": 3013,
        "deaths": 501,
        "todayDeaths": 76,
        "recovered": 786,
        "active": 22647,
        "critical": 1311,
        "casesPerOneMillion": 284,
        "deathsPerOneMillion": 6,
        "updated": 1586034307936
    };

    const props = {
        mode: 'dark' as ModeType,
        data
    };

    it('matches snapshot', () => {
        const wrapper = shallow(<EntryRow {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});