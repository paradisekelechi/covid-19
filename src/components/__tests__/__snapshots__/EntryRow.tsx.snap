// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryRow matches snapshot 1`] = `
<Styled(Styled(styled.div))
  alignment="space-evenly"
  mode="dark"
  spacing="none"
>
  <styled.div>
    <styled.img
      height="44"
      src="https://raw.githubusercontent.com/NovelCOVID/API/master/assets/flags/tr.png"
      width="100"
    />
    <Styled(styled.p)
      color="#2AA1D1"
      leftAligned={true}
    >
      Turkey
    </Styled(styled.p)>
  </styled.div>
  <styled.div>
    <Styled(styled.p)
      color="#2AA1D1"
    >
      23,934
    </Styled(styled.p)>
    <Styled(styled.p)
      color="#2AA1D1"
    >
      Total Cases
    </Styled(styled.p)>
  </styled.div>
  <styled.div>
    <Styled(styled.p)
      color="#8FB8C9"
    >
      3,013
    </Styled(styled.p)>
    <Styled(styled.p)
      color="#8FB8C9"
    >
      Cases Today
    </Styled(styled.p)>
  </styled.div>
  <styled.div>
    <Styled(styled.p)
      color="#7C5151"
    >
      501
    </Styled(styled.p)>
    <Styled(styled.p)
      color="#7C5151"
    >
      Total Deaths
    </Styled(styled.p)>
  </styled.div>
  <styled.div>
    <Styled(styled.p)
      color="#9D3535"
    >
      76
    </Styled(styled.p)>
    <Styled(styled.p)
      color="#9D3535"
    >
      Deaths Today
    </Styled(styled.p)>
  </styled.div>
  <styled.div>
    <Styled(styled.p)
      color="#488010"
    >
      786
    </Styled(styled.p)>
    <Styled(styled.p)
      color="#488010"
    >
      Recovered
    </Styled(styled.p)>
  </styled.div>
  <styled.div>
    <Styled(styled.p)
      color="#838D17"
    >
      22,647
    </Styled(styled.p)>
    <Styled(styled.p)
      color="#838D17"
    >
      Active Cases
    </Styled(styled.p)>
  </styled.div>
</Styled(Styled(styled.div))>
`;
